### PR TITLE
Parents field in documents searchable in core and front APIs

### PR DESCRIPTION
**Pull request by philipperolet**
Following #1135 

Note about the Qdrant migration: I changed locally cargo.toml to generate a bin, but won't commit it since it's a one-shot

### CODE REVIEW

File: core/src/data_sources/data_source.rs
Comment (philipperolet) [inline core/src/data_sources/data_source.rs:88]: I beleived it was important to document the new field, but not in `docs` since the field is not pushed to external builders. After thinking 5mn, I though the Document model was the right place; if there's something better, glad to move it

**Comment by philipperolet** [comment]
Currently looking at the impact of the Qdrant migration to propose a deployment plan; my early guess is that since chunks don't have a parents field in their payload yet, the operation will be swift and will not require specific downtime handling; second opinion welcome

File: core/src/data_sources/data_source.rs
Comment (spolu) [inline core/src/data_sources/data_source.rs:108]: internal id (specific to the managed data source)

File: core/src/data_sources/data_source.rs
Comment (spolu) [inline core/src/data_sources/data_source.rs:106]: ATTOW ?

File: core/src/data_sources/data_source.rs
Comment (spolu) [reply core/src/data_sources/data_source.rs:88]: Sure that's fine

File: core/src/data_sources/data_source.rs
Comment (spolu) [inline core/src/data_sources/data_source.rs:843]: let's not use `tags` as variable name here

File: core/src/stores/migrations/20230825_data_sources_qdrant_index.rs
Comment (spolu) [inline core/src/stores/migrations/20230825_data_sources_qdrant_index.rs:1]: Wooow Rust migration. Solid! :)

**Review by spolu — CHANGES_REQUESTED** [review]
LGTM! 

Tiny tiny change requested

File: core/src/data_sources/data_source.rs
Comment (philipperolet) [reply core/src/data_sources/data_source.rs:106]: At the time of writing

File: core/src/data_sources/data_source.rs
Comment (philipperolet) [reply core/src/data_sources/data_source.rs:843]: :fearful:

File: core/src/stores/migrations/20230825_data_sources_qdrant_index.rs
Comment (philipperolet) [reply core/src/stores/migrations/20230825_data_sources_qdrant_index.rs:1]: ;)

**Review by spolu — DISMISSED** [review]
LGTM assuming and good to go once we have a good understanding of the implication on production of creating the indexes :+1:

**Comment by spolu** [comment]
Note: we also have a Slack channel with QDRANT

**Comment by philipperolet** [comment]
Good to know ! I'll repost there it seems more appropriate and should get us a faster answer

**Comment by philipperolet** [comment]
I got an answer on discord. Posting it below for reference. 

TLDR: no downtime, but searches slowed down for the collection being migrated only. Hard for them to give an estimate of the time of migration

so the right plan IMHO is to migrate our own Dust slack collection and see how much it slows down searches and for how long, then decide whether 1. we can run everything smoothly 2. we better run it in the evening or 3. It's complex and we have to brainstorm more

**Comment by philipperolet** [comment]
**Index creation downtime** DISCORD THREAD
Hi Team, thanks for Qdrant! I have a question: on a prod DB, on collections of ~10k to 1M vectors (per collection), I intend to create a payload index of type keyword. The payload field represents an array of tags. How long will be the operation and will it create downtime for the online semantic search service it backs? given that  at the time, 99% of payloads do not have the field in the payload (but there will soon be a migration to populate it)
Thanks for your help 🙏 
andrey.vasnetsov — Yesterday at 7:52 PM
There are many factors which contribute into index building speed. There is no downtime, but search requests may be delayed during index building
pr — Today at 10:04 AM
Thanks a lot for your quick answer. Follow-up : if we assume that at the time of building the index, the field is not yet present in the payload, will that be speedy? What would be the other factors, other than the number of rows (here 10K to 1M) and type of field (here keyword, since the field represents an array of strings)? In general, I am wondering about the ballpark duration: a few minutes? a few  hours? or a few days? Are the search delays for the whole cluster or just for the collection being migrated? Thanks again for your help and for Qdrant in general
andrey.vasnetsov — Today at 10:08 AM
 Are the search delays for the whole cluster or just for the collection being migrated?

Just for colelction

if we assume that at the time of building the index, the field is not yet present in the payload, will that be speedy?

In general it is recommended to configure payload index before any points are created, in this case there will be firtually no delay
pr — Today at 10:39 AM
Thanks ! To be sure I understand, if the points are created, but they do not have the field in their payload yet, will it be speedy as well? 
andrey.vasnetsov — Today at 10:40 AM
In order to build index, it would need to read and parse all payload you currently have, that's the slowest part. If you have points, but do not have the field, it will only speed up the process marginally
pr — Today at 10:42 AM
Ok thanks a lot

**Comment by spolu** [comment]
Plan LGTM

**Pull request merged**
